edu.cuny.hunter.streamrefactoring.core/src/edu/cuny/hunter/streamrefactoring/core/analysis/StreamStateMachine.java

```diff
@@ -1043,133 +1051,118 @@ private void propagateStreamInstanceProperty(Collection<InstanceKey> streamInsta
 				// TODO: Can this be somehow rewritten to get blocks corresponding to terminal
 				// operations?
 				// for each call graph node in the call graph.
-				for (CGNode cgNode : engine.getCallGraph())
-					// separating client from library code, improving performance #103.
-					if (cgNode.getMethod().getDeclaringClass().getClassLoader().getReference()
-							.equals(ClassLoaderReference.Application)) {
-
-						// we can verify that only client nodes are being considered
-						LOGGER.fine(() -> "Examining client call graph node: " + cgNode);
-
-						// for each call site in the call graph node.
-						for (Iterator<CallSiteReference> callSites = cgNode.iterateCallSites(); callSites.hasNext();) {
-							// get the call site reference.
-							CallSiteReference callSiteReference = callSites.next();
-
-							// get the (declared) called method at the call site.
-							MethodReference calledMethod = callSiteReference.getDeclaredTarget();
-
-							// is it a terminal operation? TODO: Should this be
-							// cached somehow? Collection of all terminal operation
-							// invocations?
-							if (isTerminalOperation(calledMethod)) {
-								// get the basic block for the call.
```

Comment:
move line 1048 - 1065 to the method below. The rest of the code in the block was not changed.